execution_engine/src/storage/transaction_source/lmdb.rs

```diff
@@ -44,9 +89,9 @@ impl<'a> Transaction for RwTransaction<'a> {
 }
 
 impl<'a> Readable for RwTransaction<'a> {
-    fn read(&self, handle: Self::Handle, key: &[u8]) -> Result<Option<&[u8]>, Self::Error> {
+    fn read(&self, handle: Self::Handle, key: &[u8]) -> Result<Option<Bytes>, Self::Error> {
         match lmdb::Transaction::get(self, handle, &key) {
-            Ok(bytes) => Ok(Some(bytes)),
+            Ok(bytes) => Ok(Some(Bytes::from(bytes))),
```

Comment:
Is this necessary? I think we made this change with @mpapierski to avoid unnecessary allocations.

Reply:
Yes, it is. This reverted @mpapierski 's optimization as a part of the rocksdb changes:
- since we are caching the lifetimes differ
- when we switch to rocksdb this isn't possible in this way
- callsites immediately copy, in order to put values onto the reactor

Reply:
This makes sense, thanks!